homeassistant/components/sensor/srp_energy.py

```diff
@@ -0,0 +1,159 @@
+"""
+Platform for retrieving energy data from SRP.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/energy.srp/
+"""
+from datetime import datetime, timedelta
+import logging
+
+from requests.exceptions import (
+    ConnectionError as ConnectError, HTTPError, Timeout)
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_NAME, CONF_PASSWORD,
+    CONF_USERNAME, CONF_ID)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA  # noqa
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['srpenergy==1.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = "Powered by SRP Energy"
+
+DEFAULT_NAME = 'SRP Energy'
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=1440)
+ENERGY_KWH = 'kWh'
+
+ATTR_READING_COST = "reading_cost"
+ATTR_READING_TIME = 'datetime'
+ATTR_READING_USAGE = 'reading_usage'
+ATTR_DAILY_USAGE = 'daily_usage'
+ATTR_USAGE_HISTORY = 'usage_history'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_ID): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the SRP energy."""
+    _LOGGER.info("Setting up Srp Energy.")
+
+    name = config.get(CONF_NAME)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    account_id = config.get(CONF_ID)
+
+    from srpenergy.client import SrpEnergyClient
+
+    srp_client = SrpEnergyClient(account_id, username, password)
```

Comment:
We should test that credentials are ok, and otherwise log error and return from `setup_platform`.